pages/index.tsx

```diff
@@ -1,62 +1,72 @@
 import type { NextPage } from 'next'
-import { useNetwork } from 'wagmi'
+import { useAccount, useNetwork } from 'wagmi'
 import { useEffect, useState } from 'react'
-import { Button, Text, Heading, Image, SimpleGrid, Box } from '@chakra-ui/react'
+import {
+  Button,
+  Text,
+  Heading,
+  Image,
+  SimpleGrid,
+  Box,
+  Link,
+  Spacer
+} from '@chakra-ui/react'
 import { Layout } from '@/components/layouts/layout'
 import { getContractAddress } from '@/utils/contractAddress'
 import { useTotalSupply } from '@/hooks/useTotalSupply'
-import { useTokenURI } from '@/hooks/useTokenURI'
+import { useMounted } from '@/hooks/useMounted'
+import { useBalanceOf } from '@/hooks/useBalanceOf'
+import { ExternalLinkIcon } from '@chakra-ui/icons'
 
 const Home: NextPage = () => {
   const { activeChain } = useNetwork()
+  const { data } = useAccount()
   const kamonNFT = getContractAddress({
     name: 'kamonNFT',
     chainId: activeChain?.id
   })
   const openSeaTokenBaseUrl = `https://testnets.opensea.io/assets/${kamonNFT}/`
+  const { balanceOf } = useBalanceOf(kamonNFT, data?.address)
   const { totalSupply } = useTotalSupply(kamonNFT)
-  const { tokenURI } = useTokenURI(kamonNFT, Number(totalSupply))
-
-  // useState
-  const [tokenURIImage, setTokenImageURI] = useState('')
-
-  useEffect(() => {
-    if (totalSupply && tokenURI) {
-      const fetchData = async () => {
-        const pinataRequest = await fetch(tokenURI.toString())
-        const responseJson = await pinataRequest.json()
-
-        setTokenImageURI(responseJson.image)
-      }
-
-      fetchData()
-    }
-  }, [tokenURI, totalSupply])
+  const mounted = useMounted()
 
   return (
     <>
       <Layout>
         <Heading as="h2" color="white.600">
           Mint your Kamon - 家紋{' '}
         </Heading>
-        <Text m="1rem">Kamon NFT is membership of henkaku community</Text>
+        <Text m="1rem">
+          Kamon NFT is membership of henkaku community.{' '}
+          {mounted && totalSupply && (
+            <Text>🎉 {totalSupply.toString()} members minted so far 🎉</Text>
+          )}
+        </Text>
         <SimpleGrid columns={{ sm: 1, md: 1, lg: 2 }} spacing="10px">
           <div>
-            {tokenURIImage ? (
-              <Image src={tokenURIImage} alt="" />
-            ) : (
-              <Image src={'/kamonNFT_427@2x.png'} alt="" />
-            )}
+            <Image src={'/kamonNFT_427@2x.png'} alt="" />
```

Comment:
Is it like making it a fixed image once?

Reply:
for now but I will modify here in diffrent PR